cmd/ingester/app/consumer/consumer.go

```diff
@@ -100,6 +105,7 @@ func (c *Consumer) handleMessages(pc sc.PartitionConsumer) {
 	var msgProcessor processor.SpanProcessor
 
 	for msg := range pc.Messages() {
+		c.rateLimiter.await()
```

Comment:
Alternatively you could wrap `pc.Messages()` channel into another channel that limits the rate of produced messages.  Then your reading code wouldn't be exposed to the rate limiting API, and you can bypass wrapping the channel if the rate parameter is 0.

Reply:
So I thought about that. I thought it might be a performance issue to add another channel. Better idea would be to benchmark both. Will get back to you.